src/builders/webpack-config-merger.ts

```diff
@@ -0,0 +1,21 @@
+import {MergeStrategies} from './single-spa-webpack-builder-config';
+import * as webpackMerge from 'webpack-merge';
+import {Configuration} from 'webpack';
+import {differenceWith, keyBy, merge} from 'lodash'
+
+export class WebpackConfigMerger {
+    static merge(webpackConfig1: Configuration, webpackConfig2: Configuration, mergeStrategies: MergeStrategies = {}, replacePlugins = false): Configuration {
+        const mergedConfig = webpackMerge.smartStrategy(mergeStrategies)(webpackConfig1, webpackConfig2);
```

Comment:
Why aren't we just using [webpackMerge.smart()](https://github.com/survivejs/webpack-merge#smart-merging)?  It is already premade for webpack merges and in my experience works great without any configuration. It would simplify the code in this PR a fair amount.

Reply:
My aim was to afford the single-spa app developer as much control as possible over what is to be merged and how, hence the use of smartStrategy and the fact that the singleSpaWebpackConfig node in the schema is an object and not just a string. 

I'll concede that this adds a fair amount of complexity, but as I can see that there might be instances where developers might want control of how their module rules and plugins are merged, I'd rather offer that from the get go.

Reply:
Do you have a specific use case where you need the merge strategies?

I prefer starting out with something simple and only making it complex if there are strong use cases for the complexity. I've used webpack-merge a fair amount and haven't run into any issues with its smart merging.